### ISSUE TITLE
When New-Item -Path is combined with -Name, the -Path argument is inappropriately treated as a wildcard expression

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#17106]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Note: 
 * Conceptually related: #14534

The following two statements _should_ be equivalent, but aren't:

```powershell
# OK - with only a -Path argument, the path is taken *literally*
# Creates a subdir. literally named '[test]' with a file 'test.txt' in it.
New-Item -Force -Path '[test]\test.txt'

# !! BROKEN - with -Name also present, the -Path argument is
# !! interpreted as a *wildcard expression*
New-Item -Force -Path '[test]' -Name 'test.txt'
```

With `-Name` also present, `-Path` is unexpectedly treated _as a wildcard expression_.

### Expected behavior

Both commands should (re)create a subdirectory `[test]` with a file `test.txt` in it, and report a `FileInfo` instance for the latter; e.g.:

```console

    Directory: /Users/jdoe/[test]

UnixMode   User             Group                 LastWriteTime           Size Name
--------   ----             -----                 -------------           ---- ----
-rw-r--r-- jdoe             staff                4/4/2022 17:08              0 test.txt

```

### Actual behavior

The second command is situationally either a quiet no-op, or fails when wildcard pattern `[test]` matches an _existing file_ named `t` or `e` or ... or inappropriately targets an _existing directory_ named `t` or `e` ...

### Environment data

```powershell
PowerShell Core 7.3.0-preview.3
```

**Comment by jborean93** [comment]
While confusing due to the lack of `-Path` and `-LiteralPath` parameters for the `-Name` parameter set, this is documented and there is even an example https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/new-item?view=powershell-7.2#example-6-use-wildcards-to-create-files-in-multiple-directories.

**Comment by mklement0** [comment]
@jborean93, 

* Yes, that inconsistency in the documentation should be fixed.

* Separately, there's no justification for the current behavior - whether (half)-documented or not - and it should be fixed.

**Comment by jhoneill** [comment]
I think there is a use for the present behaviour, so making it more logical would be a breaking change 
```
cd $env:temp
md path1,path2,path3
new-item -type File -Path pat* -name "foo"
new-item -type File -Path "path[12345]" -name "bar"
```

In fact one might argue that 
```
new-item -type File -Path "path[12345]\baz" -force
```
Is at fault because (e.g.) `get-item "path[12345]" `  will get path1,2,3 and not path[12345] 
But then if one tries 
```
copy   4.ico -Destination  "path[12345]"
```
That goes to  path[12345]  and not 1,2 and 3 ! 

There's a good argument for making it consistent, to which the counter is going to be "what relies on present behaviour and breaks if we do ?".  

It is certainly not widely known that .Net wildcard (and therefore PowerShell ones) support `[]` in a similar way to regex or that to be sure that [ and ] work in file names they should be escaped (with ` or `` in a double quoted string).

**Comment by mklement0** [comment]
* That the parameter name is `-Path` and not `-LiteralPath` confuses the issue, but is a separate problem.

  * While renaming the parameter would be a serious breaking change, a `-LiteralPath` parameter _alias_ could be introduced.
  * `New-Item`'s `-Path` _mostly_ - and sensibly - acts like `-LiteralPath`, _except_ - confusingly - if `-Name` happens to be specified also.
    * Taking a step back: arguably, there was never a need for a separate `-Name` parameter, given the ease of, say,
        * `New-Item -Path (Join-Path '[test]' 'test.txt')`

* In the context of _creating_ files, it is confusing and unexpected to treat file paths as _wildcard expressions_. This happens or has historically happened in several places, notably also with `New-Item`'s `-Target` parameter.

  * While I can only guess, these behaviors were accidental fallouts from the implementation rather than design choices.
  *  E.g., treating the target file in `'foo' > '[2].txt'` as a wildcard makes no sense.
  * Here's the hopefully comprehensive list:
    * #14534 
    * #4726
    * #6729
    * #9225
    * #8988
    * #9202
    * #12168
  * Related problems stemming from _escaped_ representations:
    * #9475
    * #17068
    * #3725
    * #6733
    * #7999

---

As for the impact of fixing the problem:

An analogous breaking change - switching to interpretation as a _literal_ path - was considered acceptable in two related cases:

* [green-lighted, but not yet implemented] Interpret `-Target` arguments as literal paths: https://github.com/PowerShell/PowerShell/issues/13136#issuecomment-659053840
* [implemented in 7.1] #11701

In other words: there is precedent for making this - breaking, but arguably bucket-3 - change.

**Comment by SeeminglyScience** [comment]
>   * `New-Item`'s `-Path` _mostly_ - and sensibly - acts like `-LiteralPath`, _except_ - confusingly - if `-Name` happens to be specified also.
>     
>     * Taking a step back: arguably, there was never a need for a separate `-Name` parameter, given the ease of, say,
>       
>       * `New-Item -Path (Join-Path '[test]' 'test.txt')`

While I probably wouldn't say it's a pattern I would have used if I were writing it, it's not difficult to see the thought process. The actual file name is the only part that absolutely can't have wildcards. When you specify it separately, you now know that you can use wildcards to help resolve the container you'll be creating the item in.

Not my favorite way to solve this issue for sure, but interactively it can be very useful.

**Comment by mklement0** [comment]
> The actual file name is the only part that absolutely can't have wildcards.

As an aside, which points to the larger mess: The file name _can_ have wildcards, as long they're _resolved_ before trying to create the file.
While that is a terrible idea, it is how `>` / `Out-File -Path` and `Set-Content -Path` currently behave, with both complaining if there is _no_ match, and only  the former complaining in the event of _multiple_ matches, whereas the latter writes to all(!) maching files.

> interactively it can be very useful.

Interactively, it makes more sense to me to use _tab-completion_ for _up-front_ resolution, which also gives you the certainty of knowing ahead of time what specific older will actually be targeted.

In the grand scheme of things, varying the fundamental behavior of a given parameter based on whether another parameter happens to be present as well strikes me as a recipe for everlasting pain and confusion.

**Comment by jhoneill** [comment]
The cmdlet working group meets today and I may get some views there. 
I think there are two things 
1.  Use of square brackets in file names. Elsewhere they are normal characters. Windows and unix file systems have their own set of special characters which aren't allowed but square brackets are legal. For the `[` and `]` to be be meta characters and need escaping is a surprise. 
2. Inconsistent behaviour. What would be **expect** `New-Item -Force -Path 'My D*\test.txt'` to do ? error or resolve My Documents ? What if 'My D*' resolves to more than one item - error or create both ? If it doesn't resolve that's an error.  `Test[123]` should do the same thing. Different commands seem to do different things, and `[123]` isn't treated as a wild card everywhere .
